lib/build.py

```diff
@@ -265,6 +266,19 @@ def execute(self):
    def execute_(self):
       ...
 
+   def metadata_history_add(self, inst_ct):
+      hist = { "history": [ { "created": ch.now_utc_iso8601(),
+                              "created_by": "%s %s" % (self.str_name(), self.str_())}]}
+      images[image_i].metadata_append(hist)
```

Comment:
I'd recommend that the function that adds a history entry take the two? history fields directly, with names chosen by you (IMO `created` and `created_by` are confusing) e.g. `history_append(when, what)`.